### DISCUSSION TITLE
Broken pipe when sending to 127.0.0.1:20005 (VRS)

### DISCUSSION THREAD

**Discussion by jaymot123** [szpajder/dumphfdl#17] [Q&A] [answered]
I keep getting this message in the terminal when I have dumphfdl feeding VRS when set up exactly like your instructions:
```
$ dumphfdl --soapysdr driver=sdrplay --sample-rate 10000000 --system-table /home/jay/SDR_Apps/dumphfdl/etc/systable.conf --system-table-save /home/jay/SDR_Apps/systable.conf --output decoded:basestation:tcp:address=127.0.0.1,port=20005 8927 8912 8977 8921 8843 8942 8834 8939 8894 8957 8886 8885 8948
dumphfdl 1.2.0 (libacars 2.1.3)
driver=sdrplay: computed center frequency: 8905.500 kHz
System table loaded from /home/jay/SDR_Apps/dumphfdl/etc/systable.conf
Found device #0:
  driver = sdrplay
  label = SDRplay Dev0 RSP1A (SN redacted)
  serial = (SN redacted)
[INFO] devIdx: 0
[INFO] SerNo: (SN redacted)
[INFO] hwVer: 255
[INFO] rspDuoMode: 0
[INFO] tuner: 1
[INFO] rspDuoSampleFreq: 0.000000
driver=sdrplay: center frequency set to 8905.500 kHz
driver=sdrplay: frequency correction set to 0.00 ppm
driver=sdrplay: auto gain enabled
driver=sdrplay: using antenna RX
driver=sdrplay: using native sample format CS16 (full_scale: 32767.000)
[INFO] Using format CS16.
Connecting to 127.0.0.1:20005...
Connection to 127.0.0.1:20005 established
Error while sending to 127.0.0.1:20005: Broken pipe
Connection to 127.0.0.1:20005 closed
Connecting to 127.0.0.1:20005...
Connection to 127.0.0.1:20005 established
Error while sending to 127.0.0.1:20005: Broken pipe
Connection to 127.0.0.1:20005 closed
Connecting to 127.0.0.1:20005...
Connection to 127.0.0.1:20005 established
Error while sending to 127.0.0.1:20005: Broken pipe
Connection to 127.0.0.1:20005 closed
Connecting to 127.0.0.1:20005...
Connection to 127.0.0.1:20005 established
```
Feeding VRS version 3.0.0.6 mono in Linux.

**Comment by szpajder** [comment, marked as answer]
It means that VRS has terminated the connection - either due to timeout
(which is configurable in the receiver settings ) or because two (or more)
clients are trying to connect to the same VRS port at the same time.

**Comment by jaymot123** [comment]
Ah, I missed that setting to change timeout to 1800 as well as the one to unselect send keep-alive packets. Fixed. I'll test. Thank you.

**Reply by jaymot123** [reply]
By the way, I'm also feeding VRS with ADS-B from readsb on my Raspberry Pi with the standard, default port for that (I think 30005.). Should I set up a merged feed using the two receivers (which I've named ADSB and HFDL?)

**Comment by szpajder** [comment]
You may try it, but given that these feeds have different timeout values
(ie. HFDL has "Satcom" checkbox ticked which causes aircraft to stay longer
on the map) my preference is to keep them separate.

I have two ADS-B feeds and three HFDL feeds and I have two merged feeds -
first one for ADS-B, second one for HFDL.

**Reply by jaymot123** [reply]
I'm no longer getting broken pipe errors. Thanks again.
